pi_ldapproxy/bindcache.py

```diff
@@ -23,48 +23,54 @@ def __init__(self, timeout=5):
         :param timeout: Number of seconds after which the entry is removed from the bind cache
         """
         self.timeout = timeout
-        #: Map of tuples (dn, password) to insertion timestamps (determined using ``reactor.seconds``)
+        #: Map of tuples (dn, app_marker, password) to insertion timestamps (determined using ``reactor.seconds``)
         self._cache = {}
 
-    def add_to_cache(self, dn, password):
+    def add_to_cache(self, dn, app_marker, password):
```

Comment:
Hmm, good question! I'd probably go for something like ``def add_to_cache(self, cache_object, password)``, where ``cache_object=(dn, app_marker)``. Simply because it is no problem to print ``cache_object`` to the log, whereas we should never print ``password`` to the log :)